Cake.Core/Discord/Modules/HelpModule/HelpModule.cs

```diff
@@ -25,22 +25,32 @@ public HelpModule(HelpService service, CommandService commandService)
 
         [Hide]
         [Command("help")]
+        [Summary("help (searchFilter?)")]
         [Alias("cmds", "commands")]
         public async Task CommandHelp([Remainder] string command = "")
         {
             IUserMessage newMessage = await Context.Channel.SendMessageAsync("Fetching Commands...");
 
-            List<CakeEmbedBuilder> helpPages = await _service.FetchAllCommandInfoAsPages(_commandService);
-
-            await newMessage.ModifyAsync(msg => { msg.Embed = helpPages[0].Build(); msg.Content = string.Empty; });
+            List<CakeEmbedBuilder> helpPages = await _service.FetchAllCommandInfoAsPages(_commandService, command);
 
-            await newMessage.AddReactionsAsync(new Emoji[] { HelpBook.RightArrowEmoji, HelpBook.LeftArrowEmoji });
+            if (helpPages.Count <= 0)
+            {
+                await newMessage.ModifyAsync(msg => msg.Content = "There is no matching command based on the search filter!");
+            }
+            else if (helpPages.Count == 1)
+            {
+                await newMessage.ModifyAsync(msg => { msg.Embed = helpPages[0].Build(); msg.Content = string.Empty; });
+            }
+            else
+            {
+                await newMessage.ModifyAsync(msg => { msg.Embed = helpPages[0].Build(); msg.Content = string.Empty; });
 
```

Comment:
Here might be a better place to do a check if the count != 1

This way you can remove the else if you have right now